src/dataflow/src/source/s3.rs

```diff
@@ -267,15 +292,27 @@ async fn scan_bucket_task(
                 continuation_token = response.next_continuation_token;
             }
             Err(e) => {
+                let err_string = format!("{}", e);
+                tx.send(Err(S3Error::ListObjectsFailed(e)))
+                    .await
+                    .unwrap_or_else(|e| {
+                        log::debug!("unable to send error on listing objects: {}", e)
+                    });
+
                 allowed_errors -= 1;
                 if allowed_errors == 0 {
-                    log::error!("failed to list bucket {}: {}", bucket, e);
+                    log::error!("failed to list bucket {}: {}", bucket, err_string);
```

Comment:
Separately from the problem with not erroring the source above, the error message in the log does not tell you which source is the problematic one, just the bucket name:

```
2021-06-09T08:50:43.951284Z ERROR dataflow::source::s3: failed to list bucket testdrive-test-1: Error obtaining body: Error obtaining chunk: error reading a body from connection: end of file before message length reached    
```

This will make it difficult for the user to figure out which source is the problematic one based on the log alone.